### PR TITLE
Update metadata

### PR DESCRIPTION
**Pull request by XX-Yin**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
- https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/1051
- [x] "Lick sensor left" and "Lick sensor right" are not daq_names for your rig and thus can't be listed as under `daq_names`. 
      (Removed the behavior from the data stream and added the behavior to the stimulus epoch instead. And no daq names required.)
- [x] Camera names don't match the names in the rig. They must match. There is a utility function you can use to make sure that the names align between rig and session.
- [x] Reward delivery cannot be null. How are you delivery rewards without it?
 (We already have the code to generate the reward delivery, and we are waiting for the SOP to collect the reference position and reference area. )
- [x] Your `session_end_time` is before the end of one of your data streams. The end time should be the last time of all data streams and stimulus epochs.
- [x] Missing `stimulus_device_name`
- [x] These output parameters and task parameters duplicate almost the entire metadata schema. Can we please please please not do this? Very seriously, this is going to make our DocDB painful and it ultimately undermines having the schema in the first place.
- [x] eg. calibrations should be in the calibration field
- [x] frame rate should be in the device (also, check the version of the schema you're using. The cameras in your device class have a `max_frame_rate` which was changed a while ago)
 (The max_frame_rate has been removed and replaced by frame_rate)
- [x] Your `frame_num` item specifically is blocking upload to DocDB. You can't have fields with ".csv" in the name. This information should live in the quality_control class (which was only recently added to the schema, but that's where it should live) (These fields should go to the qc and they are removed here)
- [x] adding software name to the stimulus_epochs
- [x] mismatch harp names
### What issues or discussions does this update address?
resolved https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/510
### Describe the expected change in behavior from the perspective of the experimenter
No
### Describe any manual update steps for task computers
No
### Was this update tested in 446/447?
On my own computer

### CODE REVIEW

File: src/foraging_gui/GenerateMetadata.py [hunk 1]
```diff
@@ -152,17 +152,19 @@ def _mapper(self):
...
             },
-            'ephys_rig_behavior_daq_names':[
+            'rig_daq_names_janelia_detector':[
                 "Harp Behavior",
                 "Harp Sound",
                 "Harp clock synchronization board",
                 "Harp sound amplifier",
                 ],
-            'behavior_rig_behavior_daq_names':[
+            'rig_daq_names_new_detector':[
```
Comment (alexpiet) [inline src/foraging_gui/GenerateMetadata.py:161]: Can we call it "aind_detector" and not "new" since that will get confusing as time goes one

**Comment by alexpiet** [comment]
Other than the name issue, this seems good to me. Can you share a new rig.json and session.json for approval from Saskia?

Are there any other issues left to implement, or is this ready to go?

File: src/foraging_gui/GenerateMetadata.py (hunk 1 above)
Comment (XX-Yin) [reply src/foraging_gui/GenerateMetadata.py]: updated!

**Comment by XX-Yin** [comment]
> Other than the name issue, this seems good to me. Can you share a new rig.json and session.json for approval from Saskia?
> 
> Are there any other issues left to implement, or is this ready to go?

It's ready for review. I will generate a new session and rig metadata tomorrow for review

File: src/foraging_gui/GenerateMetadata.py [hunk 2]
```diff
@@ -1017,7 +1123,12 @@ def _get_behavior_software(self):
             version=f'behavior branch:{self.Obj["current_branch"]}   commit ID:{self.Obj["commit_ID"]}    version:{self.Obj["version"]}; metadata branch: {current_branch}   commit ID:{commit_ID}   version:{version}',
             url=self.Obj["repo_url"],
         ))
```
Comment (alexpiet) [inline src/foraging_gui/GenerateMetadata.py:1125]: Whats going on here? It seems like we are adding Software info to "behavior_software" and "behavior_software_origin"

**Comment by alexpiet** [comment]
@XX-Yin Can you add details to the PR about what issues were fixed in the metadata? Probably just copying the itemized list from the issue is sufficient.

File: src/foraging_gui/GenerateMetadata.py (hunk 2 above)
Comment (XX-Yin) [reply src/foraging_gui/GenerateMetadata.py:1127]: The behavior_software contains both the software version for running the behavior and the software version for generating the session metadata. The behavior_software_origin only contains the version for running the behavior. I have removed the behavior_software_origin for simplicity.

**Review by alexpiet — APPROVED** [review]
This seems like an improvement over the current metadata, so even if its not perfect, lets merge it.

**Pull request merged**